src/ride/station.c

```diff
@@ -64,7 +64,10 @@ static void ride_update_station_blocksection(rct_ride *ride, int stationIndex)
 {
 	rct_map_element *mapElement;
 
+	assert(ride != NULL);
```

Comment:
Its a precondition of the function, so perfectly valid - but can also bloat the code a great deal. It should only be done in debug mode (assert gets processed in release mode too), and more significant functions / API calls.